ledgers/ccf/Makefile

```diff
@@ -28,72 +28,100 @@ ifndef PDO_SOURCE_ROOT
 $(error Incomplete configuration, PDO_SOURCE_ROOT is not defined)
 endif
 
+CCF_COMPILE_TARGET ?= virtual
+CCF_VERSION ?= 4.0.1
+CCF_LEDGER_DIR ?= $(PDO_INSTALL_ROOT)/opt/pdo/ccf
+
 NINJA ?= ninja $(NINJA_OPTIONS)
 
 SCRIPTDIR ?= $(dir $(abspath $(lastword $(MAKEFILE_LIST))))
-CCFDSTDIR ?= $(PDO_INSTALL_ROOT)/opt/pdo/ccf
-SRCDIR ?= $(PDO_SOURCE_ROOT)
-WORKSPACEDIR := $(CCFDSTDIR)/workspace
-BINDIR := $(CCFDSTDIR)/bin/
 BLDDIR := $(SCRIPTDIR)/build
 
-PYTHON_DIR=$(CCFDSTDIR)/lib/python3.8
-
-COMPILE_TARGET=virtual
-
 all: environment install
 
-build: build-pdo-tp
+# -----------------------------------------------------------------
+# There are two environments that need to be created: the PDO
+# environment and the ledger environment. This is necessitated by
+# the incompatibilities between the client requirements (which requires
+# ccf 1.0.19 which is the last python package with the ccf client modules)
+# and the ledger requirements (which installs the python modules from
+# our current verson of CCF). The client environment may be useful on
+# any pdo installation (e.g. ledger ping test makes sense on any client)
+# while the ledger environment is only useful where ccf nodes are run
+# -----------------------------------------------------------------
+environment: pdo-environment ledger-environment
+
+pdo-environment : $(PDO_INSTALL_ROOT)
+
+ledger-environment : $(CCF_LEDGER_DIR)/lib/python3.8
+
+# This directory indicates whether the pdo python virtual
+# environment has been created
+$(PDO_INSTALL_ROOT) :
+	make -C $(PDO_SOURCE_ROOT)/build environment
+	make -C $(PDO_SOURCE_ROOT)/bin install
```

Comment:
installing all of PDO environment inside the CCF container seems to be an overkill. Doesn't hurt, I guess all that we want is ccf 1.0.19, do we need anything else ?

Reply:
i think you have a misunderstanding of how your code worked in the past. you were already installing everything in python_requirements.txt which is all this does (except cleaner).

AND...

I absolutely think we should be installing pdo client packages (and all of these scripts should be installed in every pdo client). there is a separate issue to track that discussion (#469 )